lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -199,6 +213,136 @@ static void printNoIOOp(Operation *op, OpAsmPrinter *printer) {
   printer->printOptionalAttrDict(op->getAttrs());
 }
 
+//===----------------------------------------------------------------------===//
+// spv.AccessChainOp
+//===----------------------------------------------------------------------===//
+
+static Type extractPtrTypeFromPtrToCompositeType(ArrayRef<Value *> indices,
+                                                 Type type, Location baseLoc) {
+  if (!indices.size()) {
+    return mlir::emitError(baseLoc, "'spv.AccessChain' op expected at least "
+                                    "one index for spv.AccessChain"),
+           nullptr;
+  }
+
+  auto ptrType = type.dyn_cast<spirv::PointerType>();
+  if (!ptrType) {
+    return mlir::emitError(baseLoc, "'spv.AccessChain' op expected a pointer "
+                                    "to composite type, but provided ")
+               << type,
+           nullptr;
```

Comment:
It would be better to use separate statements here given the error message takes several lines: `emitError(...); return nullptr;`. Similarly for other places in the function.

Reply:
thanks, fixed.